File: grimorio de estudos/Cs50/Problem Sets/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>



//this was much better compared to the first problem set
int main(void)
{
    //calls user input and finds length
    string input = get_string("Please, input your text here: ");
    int n = strlen(input);

    int letters = 0;
    int words = 1;
    int sentences = 0;

    //counting
    for (int i = 0; i < n; i++)
    {
            //counts letters with ASCII
            if ((input[i] >= 'a' && input[i] <= 'z') || (input[i] >= 'A' && input[i] <= 'Z'))
            {
                letters++;
            }
            //counts words with space
            else if (input[i] == ' ')
            {
                words++;
            }
            //counts sentences just like letters, with ASCII
            else if (input[i] ==  '!' || input[i] ==  '.' || input[i] ==  '?')
            {
                sentences++;
            }
        }

    //print results
    printf("%i letters\n", letters);
    printf("%i words\n", words);
    printf("%i sentences\n", sentences);

    //calculation and rounding
    float L = ((float)letters / (float)words) * 100;
    float S = ((float)sentences / (float)words) * 100;
    float calculation = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(calculation);

    //print grade
     if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

File: harvardCs50/2022/Problem Sets/sentimental-readability/readability.py
from cs50 import get_string

# Gets input from user
inputText = get_string("Please, input your text here: ")
# Length from input
textLength = len(inputText)
letters = 0
words = 1
sentences = 0

# Iterate for how many characters are there in inputText
for x in range(textLength):
    # Convert to lowercase
    text = inputText.lower()
    # Select a character from text, then gets its ASCII value
    char = ord(text[x])
    # If char is a lowercase letter
    if char >= 97 and char <= 122:
        letters += 1
    # If char is a space, add words
    elif char == 32:
        words += 1
    # If char is a . or ! or ?, add sentences
    elif char == 46 or char == 33 or char == 63:
        sentences += 1

# print(f"letters: {letters}")
# print(f"words: {words}")
# print(f"sentences: {sentences}")

# Coleman-Liau index
L = (letters / words) * 100
S = (sentences / words) * 100
calculation = 0.0588 * L - 0.296 * S - 15.8
grade = round(calculation)

# Print grade
if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")